redisinsight/api/src/modules/database-recommendation/providers/database-recommendation.provider.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { DatabaseRecommendationEntity }
+  from 'src/modules/database-recommendation/entities/database-recommendation.entity';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { plainToClass } from 'class-transformer';
+import { RecommendationDto, RecommendationsDto } from 'src/modules/database-recommendation/dto';
+import { ClientMetadata } from 'src/common/models';
+
+@Injectable()
+export class DatabaseRecommendationProvider {
+  private readonly logger = new Logger('DatabaseRecommendationProvider');
+
+  constructor(
+    @InjectRepository(DatabaseRecommendationEntity)
+    private readonly repository: Repository<DatabaseRecommendationEntity>,
+  ) {}
+
+  /**
+   * Save entire entity
+   * @param databaseId
+   * @param recommendationName
+   */
+  async create(databaseId: string, recommendationName: string): Promise<RecommendationDto> {
+    this.logger.log('Creating database recommendation');
+    return await this.repository.save(
+      plainToClass(DatabaseRecommendationEntity, { databaseId, name: recommendationName }),
+    );
+  }
+
+  /**
+   * Return list of database recommendations
+   * @param clientMetadata
+   */
+  async list(clientMetadata: ClientMetadata): Promise<RecommendationsDto> {
+    this.logger.log('Getting database recommendations list');
+    const recommendations = await this.repository
+      .createQueryBuilder('r')
+      .where({ databaseId: clientMetadata.databaseId })
+      .select(['r.id', 'r.name'])
```

Comment:
I strongly believe that you should select r.read here to be able highlight unread (new) recommendations on UI

Reply:
Right now in mockup we don't have any recommendation highlighting. Just have we new recommendations or not(field totalUnread controlling it).
BTW we have field "read" in each recommendation, so we can highlight them

Reply:
Without read field - there is no way to distinguish between old (already seen by user) and new recommendations. 
So if we will think in advance we can predict that this field will be required.
Anyway if there is no such requirement I'm totally fine with current implementation.
P.S. I assume that the same thing with createdAt field